src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -187,6 +328,305 @@ public Visitable visit(MethodCallExpr method, Void p) {
+   *
+   * @param expr the expression to be used
+   */
+  public void updateClassesFromJarSourcesForObjectCreation(ObjectCreationExpr expr) {
+    if (!isFromAJarFile(expr)) {
+      throw new RuntimeException(
+          "Check with isFromAJarFile first before using updateClassesFromJarSources");
+    }
+    String objectName = expr.getType().getName().asString();
+    ResolvedReferenceTypeDeclaration objectSolved = expr.resolve().declaringType();
+    @SuppressWarnings(
+        "signature") // this is not a precise assumption, as getClassName() will return a
+    // @FullyQualifiedName if the class is not of primitive type. However, this is
+    // favorable, since we don't have to write any additional import statements.
+    @ClassGetSimpleName String className = objectSolved.getClassName();
+    String packageName = objectSolved.getPackageName();
+    List<String> argumentsList = getArgumentsFromObjectCreation(expr);
+    UnsolvedClass missingClass = new UnsolvedClass(className, packageName);
+    UnsolvedMethod thisMethod = new UnsolvedMethod(objectName, "", argumentsList);
+    missingClass.addMethod(thisMethod);
+    this.updateMissingClass(missingClass);
+  }
+
+  /**
+   * This method checks if an expression is called by the super keyword. For example, super.visit()
+   * is such an expression.
+   *
+   * @param node the expression to be checked
+   * @return true if method is a super call
+   */
+  public static boolean isASuperCall(Expression node) {
+    if (node instanceof MethodCallExpr) {
+      Optional<Expression> caller = node.asMethodCallExpr().getScope();
+      if (caller.isEmpty()) {
+        return false;
+      }
+      return caller.get().isSuperExpr();
+    } else if (node instanceof FieldAccessExpr) {
+      Expression caller = node.asFieldAccessExpr().getScope();
+      return caller.isSuperExpr();
+    } else {
+      throw new RuntimeException("Unforeseen expression: " + node);
+    }
+  }
+
+  /**
+   * For a super call, this method will update the corresponding synthetic class
+   *
+   * @param expr the super call expression to be taken as input
+   */
+  public void updateSyntheticClassForSuperCall(Expression expr) {
+    if (!isASuperCall(expr)) {
+      throw new RuntimeException(
+          "Check if isASuperCall returns true before calling updateSyntheticClassForSuperCall");
+    }
+    if (expr instanceof MethodCallExpr) {
+      updateUnsolvedClassWithMethodCall(
+          expr.asMethodCallExpr(),
+          this.parentClass,
+          methodAndReturnType.getOrDefault(expr.asMethodCallExpr().getNameAsString(), ""));
+    } else {
+      updateUnsolvedClassWithVariables(
+          expr.asFieldAccessExpr().getNameAsString(),
+          parentClass,
+          classAndPackageMap.getOrDefault(parentClass, this.currentPackage));
+    }
+  }
+
+  /**
+   * This method will add a new variable declaration to a synthetic class. This class is mainly used
```

Comment:
In the second sentence, should "this class" be "this method"? Or did I misunderstand what you mean?